pkg/vault/test/authorizer/TimelockAuthorizer.test.ts

```diff
@@ -731,6 +734,192 @@ describe('TimelockAuthorizer', () => {
     });
   });
 
+  describe('addCanceler', () => {
+    context('when the sender is the root', () => {
+      let scheduledId = bn(0);
+      beforeEach('set sender', async () => {
+        from = root;
+      });
+
+      context('when creating canceler', () => {
+        context('for a specific scheduled execution id', () => {
+          it('can add canceler for a specific execution id', async () => {
+            expect(await authorizer.isCanceler(scheduledId, other)).to.be.false;
+
+            await authorizer.addCanceler(scheduledId, other, { from });
+
+            expect(await authorizer.isCanceler(scheduledId, other)).to.be.true;
+          });
+
+          it('emits an event', async () => {
+            const receipt = await authorizer.addCanceler(scheduledId, other, { from });
+
+            expectEvent.inReceipt(await receipt.wait(), 'CancelerAdded', { scheduledExecutionId: scheduledId });
+          });
+
+          it('cannot be add twice', async () => {
+            await authorizer.addCanceler(scheduledId, other, { from });
+
+            await expect(authorizer.addCanceler(scheduledId, other, { from })).to.be.revertedWith(
+              'ACCOUNT_IS_ALREADY_CANCELER'
+            );
+          });
+        });
+
+        context('for any scheduled execution id', () => {
+          beforeEach('set schedule id', async () => {
+            scheduledId = GLOBAL_CANCELER_SCHEDULED_EXECUTION_ID;
+          });
```

Comment:
I would not have this, and instead inline the value in order to be more explicit (similar to my suggestion above to not use `from = root`).